backbone.blazer.js

```diff
@@ -133,13 +133,10 @@ Backbone.Blazer.Router = Backbone.Router.extend({
     },
 
     _runHandler: function(handler, router, route, routeData) {
-        var result = handler.call(route, routeData);
-
-        if (result && result.redirectFragment) {
-            router.navigate(result.redirectFragment, { trigger: true });
-            return $.Deferred().reject().promise();
-        }
-
-        return $.when(result);
+        return $.when(handler.call(route, routeData)).fail(function(result) {
+            if (result && result.redirectFragment) {
+                router.navigate(result.redirectFragment, { trigger: true });
+            }
+        });
```

Comment:
Yeah, I'm gonna write some more tests. I know browser navigation stops code from executing, but I don't know if pushState does, so I want to look into that.